oak/server/oak_grpc_node.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/server/oak_grpc_node.h"
+#include "absl/memory/memory.h"
+#include "asylo/grpc/auth/enclave_server_credentials.h"
+#include "asylo/grpc/auth/null_credentials_options.h"
+#include "asylo/util/logging.h"
+#include "include/grpcpp/grpcpp.h"
+#include "oak/server/module_invocation.h"
+
+namespace oak {
+
+std::unique_ptr<OakGrpcNode> OakGrpcNode::Create() {
+  std::unique_ptr<OakGrpcNode> node = absl::WrapUnique(new OakGrpcNode());
+  node->grpc_in_ = std::make_shared<MessageChannel>();
+  node->grpc_out_ = std::make_shared<MessageChannel>();
+
+  // Build Server
+  grpc::ServerBuilder builder;
+
+  // Use ":0" notation so that server listens on a free port.
+  builder.AddListeningPort(
+      "[::]:0", asylo::EnclaveServerCredentials(asylo::BidirectionalNullCredentialsOptions()),
+      &node->port_);
+  builder.RegisterService(node.get());
+
+  // Add a completion queue and a generic service, in order to proxy incoming RPCs to the Oak Node.
+  node->completion_queue_ = builder.AddCompletionQueue();
+
+  // Register an async service.
+  node->module_service_ = absl::make_unique<grpc::AsyncGenericService>();
+  builder.RegisterAsyncGenericService(node->module_service_.get());
+
+  node->server_ = builder.BuildAndStart();
+  if (!node->server_) {
+    grpc::Status(grpc::INTERNAL, "Failed to start gRPC server");
```

Comment:
`grpc::Status` what? :) Did you mean to return it or something? The current return type is not even compatible anyways, at best you could log and return `nullptr` here I guess.